src/core/lib/event_engine/thread_pool.cc

```diff
@@ -85,7 +93,10 @@ ThreadPool::ThreadPool(int reserve_threads)
   }
 }
 
-ThreadPool::~ThreadPool() { state_->queue.SetShutdown(); }
+ThreadPool::~ThreadPool() {
+  state_->queue.SetShutdown();
+  state_->thread_count.BlockUntilThreadCount(g_threadpool_thread ? 1 : 0);
```

Comment:
Could you add a comment on why one thread may still exist after the the thread pool is destroyed

Reply:
fixed - also improved the test so that this matters!